python/cudf/cudf/tests/test_indexing.py

```diff
@@ -99,16 +100,35 @@ def pdf_gdf_multi():
         + ["numpy.array[%s]" % np.dtype(t).type.__name__ for t in index_dtypes]
     ),
 )
-def test_series_indexing(i1, i2, i3):
+@pytest.mark.parametrize(
+    "arg",
+    [
+        1,
+        -1,
+        "b",
+        np.int32(1),
+        np.uint32(1),
+        np.int8(1),
+        np.uint8(1),
+        np.int16(1),
+        np.uint16(1),
+        np.int64(1),
+        np.uint64(1),
+    ],
+)
+def test_series_indexing(i1, i2, i3, arg):
```

Comment:
Sorry, I realized `test_series_indexing` here to too systematic to fit in your test case. Maybe you should create a separate test case called `test_series_get_item_iloc_defer` for your case and place next to https://github.com/rapidsai/cudf/blob/9617fcd452a66bdeb9c2275216e866b78f997508/python/cudf/cudf/tests/test_indexing.py#L140